demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -0,0 +1,1051 @@
+                                                                           xDeviceAverageTemperature, sampleazureiotDOUBLE_DECIMAL_PLACE_DIGITS ) )
+             != eAzureIoTSuccess )
+    {
+        LogError( ( "Error appending average temp: result 0x%08x", xResult ) );
+    }
+    else if( ( xResult = AzureIoTJSONWriter_AppendPropertyWithStringValue( pxWriter, ( const uint8_t * ) sampleazureiotCOMMAND_START_TIME,
+                                                                           strlen( sampleazureiotCOMMAND_START_TIME ),
+                                                                           ucCommandStartTimeValueBuffer, ulSinceTimeLength ) )
+             != eAzureIoTSuccess )
+    {
+        LogError( ( "Error appending start time: result 0x%08x", xResult ) );
+    }
+    /* Faking the end time to simplify dependencies on <time.h> */
+    else if( ( xResult = AzureIoTJSONWriter_AppendPropertyWithStringValue( pxWriter, ( const uint8_t * ) sampleazureiotCOMMAND_END_TIME,
+                                                                           strlen( sampleazureiotCOMMAND_END_TIME ),
+                                                                           ( const uint8_t * ) sampleazureiotCOMMAND_FAKE_END_TIME,
+                                                                           strlen( sampleazureiotCOMMAND_FAKE_END_TIME ) ) )
+             != eAzureIoTSuccess )
+    {
+        LogError( ( "Error appending end time: result 0x%08x", xResult ) );
+    }
+    else if( ( xResult = AzureIoTJSONWriter_AppendEndObject( pxWriter ) )
+             != eAzureIoTSuccess )
+    {
+        LogError( ( "Error appending end object: result 0x%08x", xResult ) );
+    }
+
+    return xResult;
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Command message callback handler
+ */
+static void prvHandleCommand( AzureIoTHubClientCommandRequest_t * pxMessage,
+                              void * pvContext )
+{
+    AzureIoTResult_t xResult;
+    AzureIoTJSONReader_t xReader;
+    AzureIoTJSONWriter_t xWriter;
+    int32_t lCommandNameLength;
+    int32_t ulCommandPayloadLength;
+    AzureIoTHubClient_t * pxHandle = ( AzureIoTHubClient_t * ) pvContext;
+
+    LogInfo( ( "Command payload : %.*s \r\n",
+               pxMessage->ulPayloadLength,
+               pxMessage->pvMessagePayload ) );
+
+    lCommandNameLength = strlen( sampleazureiotCOMMAND_MAX_MIN_REPORT );
+
+    if( ( lCommandNameLength == pxMessage->usCommandNameLength ) &&
+        ( strncmp( sampleazureiotCOMMAND_MAX_MIN_REPORT, ( const char * ) pxMessage->pucCommandName, lCommandNameLength ) == 0 ) )
+    {
+        /* Is for max min report */
+        xResult = AzureIoTJSONReader_Init( &xReader, pxMessage->pvMessagePayload, pxMessage->ulPayloadLength );
+        configASSERT( xResult == eAzureIoTSuccess );
+
+        xResult = AzureIoTJSONWriter_Init( &xWriter, ucCommandPayloadBuffer, sizeof( ucCommandPayloadBuffer ) );
```

Comment:
If I've never seen this pattern before - which I probably haven't if I'm using the sample - this is going to take some review for me to get my head around.

Even though not strictly necessary to someone knowing all this, I recommend a comment on this one specifically like
/* Initialize xWriter, which will be used to build the JSON response buffer, with a buffer for it to write into. */